sn_networking/src/lib.rs

```diff
@@ -1111,6 +1121,76 @@ impl Network {
         Ok(closest_peers.into_iter().cloned().collect())
     }
 
+    /// Returns the closest peers to the given `XorName`, sorted by their distance to the xor_name.
+    /// If `client` is false, then include `self` among the `closest_peers`
+    /// Returns all peers found inside the range
+    ///
+    /// If less than CLOSE_GROUP_SIZE peers are found, it will return all the peers found up to the CLOSE_GROUP_SIZE
+    pub async fn get_all_close_peers_in_range_or_close_group(
+        &self,
+        key: &NetworkAddress,
+        client: bool,
+    ) -> Result<Vec<PeerId>> {
+        let pretty_key = PrettyPrintKBucketKey(key.as_kbucket_key());
+        debug!("Getting the all closest peers in range of {pretty_key:?}");
+        let (sender, receiver) = oneshot::channel();
+        self.send_network_swarm_cmd(NetworkSwarmCmd::GetClosestPeersToAddressFromNetwork {
+            key: key.clone(),
+            sender,
+        });
+
+        let found_peers = receiver.await?;
+
+        // Count self in if among the CLOSE_GROUP_SIZE closest and sort the result
+        let result_len = found_peers.len();
+        let mut closest_peers = found_peers;
+
+        let expected_range = self.get_range().await?;
+
+        // ensure we're not including self here
+        if client {
+            // remove our peer id from the calculations here:
+            closest_peers.retain(|&x| x != self.peer_id());
+            if result_len != closest_peers.len() {
+                info!("Remove self client from the closest_peers");
+            }
+        }
+
+        if tracing::level_enabled!(tracing::Level::DEBUG) {
+            let close_peers_pretty_print: Vec<_> = closest_peers
+                .iter()
+                .map(|peer_id| {
+                    format!(
+                        "{peer_id:?}({:?})",
+                        PrettyPrintKBucketKey(NetworkAddress::from_peer(*peer_id).as_kbucket_key())
+                    )
+                })
+                .collect();
+
+            debug!(
+                "Network knowledge of closest peers to {pretty_key:?} are: {close_peers_pretty_print:?}"
+            );
+        }
+
+        let mut restricted_closest_peers =
+            sort_peers_by_address_and_limit_by_distance(&closest_peers, key, expected_range)?;
+
+        if restricted_closest_peers.len() < CLOSE_GROUP_SIZE {
+            warn!(
+                    "Getting close peers to {pretty_key:?} current GetRange of {:?} too strict giving insufficient peers... Falling back to all peers found"
+                , expected_range.ilog2());
+
+            restricted_closest_peers =
+                sort_peers_by_address_and_limit(&closest_peers, key, CLOSE_GROUP_SIZE)?;
+        }
+
+        debug!(
+            "Network knowledge of closest peers in range of {key:?} are: {:?}",
```

Comment:
the log better to be : 
Network knowledge of closest peers in range of {:?} to target {key:?} are: {:?}, expected_range.log2(), restricted_closest_peers.len()